### ISSUE TITLE
g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

### ISSUE THREAD

**Issue by xor256** [wxWidgets/wxWidgets#22619]
sample/concole.cpp  **wx 3.1.7**

when compile it with
```
$g++ -Wall -pedantic -pedantic-errors `wx-config-3.0 --cxxflags --unicode` -DwxUSE_GUI=0 `wx-config-3.0 --cxxflags --unicode` `wx-config-3.0 --libs --unicode` console.cpp  -o co
```
and run I get

```
./co
Welcome to the wxWidgets 'console' sample!
For more information, run it again with the --help option

(process:6900): GLib-GObject-WARNING **: 09:49:18.971: invalid (NULL) pointer instance

(process:6900): GLib-GObject-CRITICAL **: 09:49:18.971: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(process:6900): GLib-GObject-WARNING **: 09:49:18.971: invalid (NULL) pointer instance

(process:6900): GLib-GObject-CRITICAL **: 09:49:18.971: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

```

where is different notma compiling and my compiling?

P.S. I cant compile   **wxCMD_LINE_HIDDEN** today is     **wxCMD_LINE_NONE**

**Comment by vadz** [comment]
Are you compiling the sample from 3.1.7 using wx-config from 3.0 intentionally? This also explains your error due to the use of `wxCMD_LINE_HIDDEN` which doesn't exist in 3.0.

Anyhow, I can't reproduce the problem when building the sample in 3.2.0 normally, so I strongly suspect some local problem. But if not, please try debugging the messages above by running the sample under gdb and breaking on `g_log_writer_standard_streams` and then use `bt` to see where these messages come from.

**Comment by xor256** [comment]
no problem ;-) reproduce

```
$uname -a
Linux localhost.localdomain 5.11.22-100.fc32.x86_64 #1 SMP Wed May 19 18:58:25 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

```
$wx-config-3.0 --version
3.0.4
```

just type make
https://github.com/xor256/error-cl

**Comment by vadz** [comment]
Is it related to setting Polish locale? I don't have it on my system, so the output is:
```
19:53:14: Warning: Cannot set locale to language "Polish".
start [0]
OK
stop [0]
ilo?? argument?w programu './a.out': 1
```

but changing it to `wxLANGUAGE_FRENCH` makes the program run successfully.

Please try debugging this (compile and link with `-g`) and try putting a breakpoint on `g_log_writer_standard_streams` to at least see where are these messages coming from.

**Comment by xor256** [comment]
How?

**Comment by vadz** [comment]
> 1. You not type 'make' why? this is important because i show how compile it

I've executed exactly the same `g++` command you've shown. In any case, this shouldn't have any effect.
 
>     2. How? using debug?

Yes:

1. Compile and link with `-g`.
2. Run `gdb ./cl`
3. Do `start`, then `b g_log_writer_standard_streams`, then `continue`.
4. When gdb stops on the breakpoint, type `bt` to see where the function is called from.

**Comment by xor256** [comment]
1. You run ./a.out not ./cl ;-) therefore i think You not use my make ;-)
2. i must download 5GiB packages to run this ......

**Comment by 0xtrzy** [comment]
yes, I have this same eror from repo example

```
start [0]
OK
stop [0]
ilość argumentów programu './cl': 1

(process:5228): GLib-GObject-WARNING **: 19:09:35.627: invalid (NULL) pointer instance

(process:5228): GLib-GObject-CRITICAL **: 19:09:35.628: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(process:5228): GLib-GObject-WARNING **: 19:09:35.628: invalid (NULL) pointer instance

(process:5228): GLib-GObject-CRITICAL **: 19:09:35.628: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
```

fedora 32 x86
wx 3.0.4

**Comment by vadz** [comment]
Thanks, as @xor256 edited their comment I didn't notice the stack.

Looking at it, it's immediately clear that this was fixed back in 17de4e31fb (Avoid using NULL GtkSettings pointer, 2017-11-25) which was backported into 3.0 branch as 904c32ce852fd9bc8449c35aab45b2109da79784 and is part of 3.0.5.

So this bug was fixed a long time ago, I don't know why does Fedora still have 3.0.4. And, of course, 3.1.7 in the original bug report was wrong, as this bug was fixed in 3.1 back in 3.1.1.

**Comment by swt2c** [comment]
> So this bug was fixed a long time ago, I don't know why does Fedora still have 3.0.4. And, of course, 3.1.7 in the original bug report was wrong, as this bug was fixed in 3.1 back in 3.1.1.

Fedora 32 was EOL on 2021-05-25.  @xor256, I would strongly recommend that you upgrade to a supported Fedora release, 35 or newer.

Supported Fedoras have 3.0.5.1 and some variant of 3.1.5+.
